### ISSUE TITLE
Grid recreate

### ISSUE THREAD

**Issue by olegrb** [JamesRandall/Simple-Paging-Grid#14]
Hello,
Every time I refresh the grid's data source I have to refresh its internal structure (headers, templates, etc.)
Due to the fact that internal structure refresh is not supported I do it in the following manner:
 $("#tableDiv").simplePagingGrid({
                columnNames: columnNames,
                columnKeys: columnKeys,
                columnWidths: columnWidths,
                headerTemplates: headerTemplates,
                showGotoPage: false,

```
            data: {
                currentPage: currData,
                totalRows: currData.length
            }
        });
```

But on the second run it fails with the following error:  
Unhandled exception at line 757, column 17 in http://localhost:51836/js/simplePagingGrid-0.5.0.2.js
0x800a138f - Microsoft JScript runtime error: Unable to get value of the property 'apply': object is null or undefined

I would like to know what is the best way to re-create the SimplePaging grid.

Thanks in Advance
Best Regards
Oleg

**Comment by JamesRandall** [comment]
I think thats a bug - I added jquery UI style methods a couple of releases back and frankly never considered recreating the grid. I'll get that fixed.

I'm hoping to get onto the grid at the weekend.

Thanks for taking the time to report this and provide all the details you have on the other issue. Really appreciate it.

**Comment by olegrb** [comment]
And I want to thank you for developing this fine piece of software !

**Comment by dcartistic** [comment]
James -

I had the same issue, but I was able to resolve it, so I'm not sure it's a bug.  In my app, the page using the simple grid loads all the data into the grid when the user accesses the page for the first time, and the page "pushes" all the data into the rows array (ie: rows.push).  The page has a search interface, and what was happening was that the grid would load without any issues the first time, but if I clicked the search button, I would get the error Oleg mentioned.  So, what I did was add the following code to the search button click event:

if (rows.length > 0) {
                     rows.length = 0;

```
             }
```

In addition, the method that loads the data -- SearchApplications -- has a parameter called dataRefresh, which can be true or false.  When the page loads for the first time, dataRefresh is false and the method "pushes" that data into the rows array.  When the user clicks the search button, the SearchApplications method is called, passing dataRefresh as true.  If dataRefresh is true, then the method splices -- instead of pushing -- the data into the rows array.  In addition, it then calls the simplePagingGrid method ( $("#exampleGrid").simplePagingGrid("refresh", rows);), instead of initializing the grid, like so:

if (!dataRefresh) {

```
                      $("#exampleGrid").simplePagingGrid({
                          columnNames: ["ID", "Name", "Address", "Phone Number", "Status"],
                          columnKeys: ["ID", "Name", "Address", "PhoneNumber", "Status"],
                          columnWidths: ["10%", "20%", "30%", "10%", "10%"],
                          sortable: [true, true, true, true, true],
                          initialSortColumn: "Name",

                          data: {
                              currentPage: rows,
                              totalRows: rows.length
                          }

                      });
                  }
                  else {
                      $("#exampleGrid").simplePagingGrid("refresh", rows);
                  }
```

Maybe there's a better way to do this.  Anyway, this solution fixed the error.

Craig

**Comment by JamesRandall** [comment]
Thanks for the feedback. Craig: I've used code like your helpful example myself when creating the grid on the first search and that is the way to do it.

I'll need to look into it though (struggling to find time at the moment) as it certainly shouldn't crash if you call the initialiser twice. If nothing else a bit of robustness improvement perhaps.

**Comment by JamesRandall** [comment]
There was an issue with this and I've fixed it in v0.6.0.0 and added an example that shows how to recreate the grid.
